agent/agent.go

```diff
@@ -2052,6 +2047,14 @@ func (a *Agent) AddCheck(check *structs.HealthCheck, chkType *structs.CheckType,
 	a.checkLock.Lock()
 	defer a.checkLock.Unlock()
 
+	// snapshot the current state of the health check to avoid potential flapping
+	existing := a.State.Check(check.CheckID)
+	defer func() {
+		if existing != nil {
+			a.State.UpdateCheck(check.CheckID, existing.Status, existing.Output)
+		}
+	}()
+
```

Comment:
It's not obvious to me why this is different from the old code that takes a snapshot before any of this and then restores all checks in that snapshot afterwards?